### PR TITLE
feature(whisper): preview PR of loading all remote openai whisper models.

### PR DESCRIPTION
**Pull request by crutcher**
feat(cli): name Whisper models in `whisper-cli`: prefabs, a pretrained index, and a digest-pinned weights cache

`transcribe --model openai/tiny.en` (or `large`, `turbo`, or a checkpoint path) now loads any of the twelve checkpoints `openai-whisper` ships, as `whisper.load_model` names them. The default, `openai/base`, resolves to the checkpoint `bunsen-bundled-whisper` already fetches, so the existing behaviour and its no-network build are unchanged.

The index lives in `examples/whisper-cli/src/models/`, deliberately not in bunsen: it is a rough-in of the shape a `kits::speech::whisper::pretrained` index could take, worked out against real files first. It keeps three things apart, because they vary independently:

* A prefab is a public configuration with no weights: the geometry (`n_mels`, vocabulary size, `d_model`, contexts, layers). The table is upstream's `ModelDimensions`, typed in, as a `StaticPreFabMap` over `WhisperApiConfig`, bunsen's existing prefab type (`PREFAB_RESNET_MAP`). The vocabulary size is part of it: `tiny` and `tiny.en` are different prefabs; `large-v1` and `large-v2` share `large`; `large-v3` has 128 mels and a hundredth language; `large-v3-turbo` a four-layer decoder.
* A pretrained is a trained set of weights for one prefab, with a format (the quantization axis; OpenAI ships one, fp16 PyTorch), a SHA-256, the aliases upstream honours (`large`, `turbo`), and an ordered list of sources: the bundled file, `openai-whisper`'s own `~/.cache/whisper`, and the URL. The `openai` table is generated from `_MODELS`.
* The weights cache puts files at `<cache>/whisper/<provider>/<sha256>/ <file>` under `BunsenDiskCache`'s directory. The digest in the path is the pin, as in upstream's URLs: a file there was verified when written and is trusted after, and a re-pinned model cannot collide with a stale one. Downloads stream to a `.partial`, hash as they land, and rename only on a match; a file in upstream's cache is hashed once and linked in; the bundled file is used in place.

Before any tensors are read, the checkpoint is scanned and its geometry checked against the prefab its name promised, so a wrong file under a right name is an error rather than a mis-shaped model. A `models` subcommand (`list`, `prefabs`, `fetch --verify`, `inspect`) exposes the index, with `--cache-dir`, `--offline` and `--upstream-cache-dir` shared with `transcribe`.

Verified on every entry: all twelve checkpoints were fetched (or read from the bundled file and from upstream's cache), scanned as their prefab, and loaded on wgpu to transcribe a clip. `cargo test -p whisper-cli` pins the table to the bundled `base.pt`, and the resolver, the cache layout, the offline and mismatch paths, and the digest code.

Two gaps in bunsen the rough-in works around, to lift with it: `StaticPretrainedWeightsDescriptor` has no digest, format or local source, so the prefab map carries `weights: None`; and `BunsenDiskCache::load_cached_path` has no hash and drops the per-file result of a download, so a 404 returns `Ok` with nothing on disk. The disk cache is used here only to resolve the directory.

New dependencies for `whisper-cli` alone, all already in the workspace table: `ureq`, `sha2`, `indicatif`, and `tempfile` for tests.

### CODE REVIEW

**Comment by crutcher** [comment]
```terminaloutput
 % cargo run --release -p whisper-cli --features bunsen/metal -- transcribe --model openai/turbo  crates/validation/whisper-model-validation/testdata/jfk_moon.mp3 -q
the moon. Why choose this as our goal? And they may well ask, why climb the highest mountain? Why, 35 years ago, fly the Atlantic? Why does Rice play Texas? We choose to go to the moon. We choose to go to the moon. We choose to go to the moon in this decade and do the other things,
not because they are easy, but because they are hard. Because that goal will serve to organize and measure the best of our energies and skills. Because that challenge is one that we're willing to accept, one we are unwilling to postpone, and one we intend to win, and the others too. It is for these reasons
```
